electrum/wallet.py

```diff
@@ -754,10 +754,13 @@ def create_invoice(self, *, outputs: List[PartialTxOutput], message, pr, URI) ->
         height=self.get_local_height()
         if pr:
             return OnchainInvoice.from_bip70_payreq(pr, height)
-        if '!' in (x.value for x in outputs):
-            amount = '!'
-        else:
-            amount = sum(x.value for x in outputs)
+        amount = 0
+        for x in outputs:
+            if check_max_spend(x.value):
```

Comment:
If using the changed `parse_max_spend` I suggest in another comment,
you could instead do `if parse_max_spend(x.value):` here